store/v2/commitment/store.go

```diff
@@ -33,20 +41,24 @@ type CommitStore struct {
 	logger     corelog.Logger
 	metadata   *MetadataStore
 	multiTrees map[string]Tree
+
+	mountTreeFn MountTreeFn
+	oldTrees    sync.Map
 }
 
 // NewCommitStore creates a new CommitStore instance.
-func NewCommitStore(trees map[string]Tree, db corestore.KVStoreWithBatch, logger corelog.Logger) (*CommitStore, error) {
+func NewCommitStore(trees map[string]Tree, db corestore.KVStoreWithBatch, mountTreeFn MountTreeFn, logger corelog.Logger) (*CommitStore, error) {
```

Comment:
I'm not sure about this API change.. `mountTreeFn MountTreeFn` in this constructor feels off from a DevEx perspective.  Can it be defined internally in CommitStore?